// react-bootstrap-typeahead/es/core/Context.js
import _inheritsLoose from "@babel/runtime/helpers/inheritsLoose";
import _extends from "@babel/runtime/helpers/extends";
import pick from "lodash/pick";
import createReactContext from 'create-react-context';
import React from 'react';

/* istanbul ignore next */
var arrToObj = function arrToObj(keys) {
  return keys.reduce(function (obj, k) {
    var _extends2;

    return _extends({}, obj, (_extends2 = {}, _extends2[k] = undefined, _extends2));
  }, {});
};
/**
 * $Pick utility
 * https://github.com/facebook/flow/issues/3367
 */

/* eslint-disable flowtype/generic-spacing */


/* eslint-enable flowtype/generic-spacing */
export var TypeaheadContextKeys = ['activeIndex', 'id', 'initialItem', 'onActiveItemChange', 'onAdd', 'onInitialItemChange', 'onMenuItemClick', 'selectHintOnEnter'];
export var TypeaheadContext = createReactContext({});
export var InputContextKeys = ['activeIndex', 'disabled', 'id', 'inputProps', 'inputRef', 'isFocused', 'isMenuShown', 'labelKey', 'multiple', 'onBlur', 'onChange', 'onFocus', 'onKeyDown', 'onRemove', 'placeholder', 'selected'];
export var InputContext = createReactContext({});
export var MenuContextKeys = ['align', 'dropup', 'flip', 'id', 'labelKey', 'onMenuToggle', 'positionFixed', 'referenceElement', 'results', 'text'];
export var MenuContext = createReactContext({});
export var withContext = function withContext(Component, values) {
  // Note: Use a class instead of function component to support refs.

  /* eslint-disable-next-line react/prefer-stateless-function */
  return (
    /*#__PURE__*/
    function (_React$Component) {
      _inheritsLoose(_class, _React$Component);

      function _class() {
        return _React$Component.apply(this, arguments) || this;
      }

      var _proto = _class.prototype;

      _proto.render = function render() {
        var _this = this;

        return React.createElement(TypeaheadContext.Consumer, null, function (context) {
          return React.createElement(Component, _extends({}, _this.props, pick(context, values)));
        });
      };

      return _class;
    }(React.Component)
  );
};
