// @patternfly/react-table/dist/js/components/Table/utils/decorators/cellActions.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactStyles = require('@patternfly/react-styles');

var _tableCss = require('../../../../@patternfly/patternfly/components/Table/table.css.js');

var _tableCss2 = _interopRequireDefault(_tableCss);

var _ActionsColumn = require('../../ActionsColumn');

var _ActionsColumn2 = _interopRequireDefault(_ActionsColumn);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = function (actions) {
  return function (label, _ref) {
    var rowIndex = _ref.rowIndex,
        _ref$column$extraPara = _ref.column.extraParams,
        dropdownPosition = _ref$column$extraPara.dropdownPosition,
        dropdownDirection = _ref$column$extraPara.dropdownDirection;
    return {
      className: (0, _reactStyles.css)(_tableCss2.default.tableAction),
      isVisible: true,
      children: _react2.default.createElement(
        _ActionsColumn2.default,
        {
          items: actions,
          dropdownPosition: dropdownPosition,
          dropdownDirection: dropdownDirection,
          rowId: rowIndex
        },
        label
      )
    };
  };
};
