// @patternfly/react-table/dist/js/components/Table/utils/decorators/collapsible.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.expandedRow = exports.collapsible = undefined;

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactStyles = require('@patternfly/react-styles');

var _tableCss = require('../../../../@patternfly/patternfly/components/Table/table.css.js');

var _tableCss2 = _interopRequireDefault(_tableCss);

var _CollapseColumn = require('../../CollapseColumn');

var _CollapseColumn2 = _interopRequireDefault(_CollapseColumn);

var _ExpandableRowContent = require('../../ExpandableRowContent');

var _ExpandableRowContent2 = _interopRequireDefault(_ExpandableRowContent);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var collapsible = exports.collapsible = function collapsible(value, _ref) {
  var rowIndex = _ref.rowIndex,
      rowData = _ref.rowData,
      _ref$column$extraPara = _ref.column.extraParams,
      onCollapse = _ref$column$extraPara.onCollapse,
      _ref$column$extraPara2 = _ref$column$extraPara.rowLabeledBy,
      rowLabeledBy = _ref$column$extraPara2 === undefined ? 'simple-node' : _ref$column$extraPara2,
      _ref$column$extraPara3 = _ref$column$extraPara.expandId,
      expandId = _ref$column$extraPara3 === undefined ? 'expand-toggle' : _ref$column$extraPara3;

  function onToggle(event) {
    onCollapse && onCollapse(event, rowIndex, rowData && !rowData.isOpen);
  }
  return {
    className: (0, _reactStyles.css)(_tableCss2.default.tableToggle),
    isVisible: true,
    children: _react2.default.createElement(
      _CollapseColumn2.default,
      {
        'aria-labelledby': '' + rowLabeledBy + rowIndex + ' ' + expandId + rowIndex,
        onToggle: onToggle,
        id: expandId + rowIndex,
        isOpen: rowData && rowData.isOpen
      },
      value
    )
  };
};

var expandedRow = exports.expandedRow = function expandedRow(colSpan) {
  var expandedRowFormatter = function expandedRowFormatter(value, _ref2) {
    var rowIndex = _ref2.rowIndex,
        rowData = _ref2.rowData,
        _ref2$column$extraPar = _ref2.column.extraParams.contentId,
        contentId = _ref2$column$extraPar === undefined ? 'expanded-content' : _ref2$column$extraPar;
    return rowData.hasOwnProperty('parent') && {
      colSpan: colSpan,
      children: _react2.default.createElement(
        _ExpandableRowContent2.default,
        { id: contentId + rowIndex },
        value.title || value
      )
    };
  };
  return expandedRowFormatter;
};
