// react-bootstrap-typeahead/es/components/Typeahead.react.js
import _assertThisInitialized from "@babel/runtime/helpers/assertThisInitialized";
import _inheritsLoose from "@babel/runtime/helpers/inheritsLoose";
import _defineProperty from "@babel/runtime/helpers/defineProperty";
import _extends from "@babel/runtime/helpers/extends";
import cx from 'classnames';
import isFunction from "lodash/isFunction";
import PropTypes from 'prop-types';
import React from 'react';
import Typeahead from '../core/Typeahead';
import ClearButton from './ClearButton.react';
import Loader from './Loader.react';
import TypeaheadInputMulti from './TypeaheadInputMulti.react';
import TypeaheadInputSingle from './TypeaheadInputSingle.react';
import TypeaheadMenu from './TypeaheadMenu.react';
import { preventInputBlur } from '../utils';
var propTypes = {
  /**
   * Specifies the size of the input.
   */
  bsSize: PropTypes.oneOf(['large', 'lg', 'small', 'sm']),

  /**
   * Displays a button to clear the input when there are selections.
   */
  clearButton: PropTypes.bool,

  /**
   * Bootstrap 4 only. Adds the `is-invalid` classname to the `form-control`.
   */
  isInvalid: PropTypes.bool,

  /**
   * Indicate whether an asynchronous data fetch is happening.
   */
  isLoading: PropTypes.bool,

  /**
   * Bootstrap 4 only. Adds the `is-valid` classname to the `form-control`.
   */
  isValid: PropTypes.bool,

  /**
   * Callback for custom input rendering.
   */
  renderInput: PropTypes.func,

  /**
   * Callback for custom menu rendering.
   */
  renderMenu: PropTypes.func
};
var defaultProps = {
  clearButton: false,
  isInvalid: false,
  isLoading: false,
  isValid: false,
  renderMenu: function renderMenu(results, menuProps) {
    return (// TODO: Merged props not registering correctly.
      // $FlowFixMe
      React.createElement(TypeaheadMenu, _extends({}, menuProps, {
        options: results
      }))
    );
  }
};

var TypeaheadComponent =
/*#__PURE__*/
function (_React$Component) {
  _inheritsLoose(TypeaheadComponent, _React$Component);

  function TypeaheadComponent() {
    var _this;

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _React$Component.call.apply(_React$Component, [this].concat(args)) || this;

    _defineProperty(_assertThisInitialized(_this), "_instance", null);

    _defineProperty(_assertThisInitialized(_this), "getInstance", function () {
      return _this._instance;
    });

    _defineProperty(_assertThisInitialized(_this), "_renderInput", function (inputProps) {
      var _this$props = _this.props,
          bsSize = _this$props.bsSize,
          isInvalid = _this$props.isInvalid,
          isValid = _this$props.isValid,
          multiple = _this$props.multiple,
          renderInput = _this$props.renderInput,
          renderToken = _this$props.renderToken;

      if (isFunction(renderInput)) {
        return renderInput(inputProps);
      }

      var props = _extends({}, inputProps, {
        bsSize: bsSize,
        isInvalid: isInvalid,
        isValid: isValid
      });

      return multiple ? React.createElement(TypeaheadInputMulti, _extends({}, props, {
        renderToken: renderToken
      })) : React.createElement(TypeaheadInputSingle, props);
    });

    _defineProperty(_assertThisInitialized(_this), "_renderMenu", function (results, menuProps) {
      var _this$props2 = _this.props,
          emptyLabel = _this$props2.emptyLabel,
          maxHeight = _this$props2.maxHeight,
          newSelectionPrefix = _this$props2.newSelectionPrefix,
          renderMenu = _this$props2.renderMenu,
          renderMenuItemChildren = _this$props2.renderMenuItemChildren;
      return renderMenu(results, _extends({}, menuProps, {
        emptyLabel: emptyLabel,
        maxHeight: maxHeight,
        newSelectionPrefix: newSelectionPrefix,
        renderMenuItemChildren: renderMenuItemChildren
      }));
    });

    _defineProperty(_assertThisInitialized(_this), "_renderAux", function (props) {
      var bsSize = props.bsSize,
          clearButton = props.clearButton,
          disabled = props.disabled,
          isLoading = props.isLoading,
          onClear = props.onClear,
          selected = props.selected;
      var content;

      if (isLoading) {
        content = React.createElement(Loader, {
          bsSize: bsSize
        });
      } else if (clearButton && !disabled && selected.length) {
        content = React.createElement(ClearButton, {
          bsSize: bsSize,
          onClick: onClear,
          onFocus: function onFocus(e) {
            // Prevent the main input from auto-focusing again.
            e.stopPropagation();
          },
          onMouseDown: preventInputBlur
        });
      }

      return content ? React.createElement("div", {
        className: cx('rbt-aux', {
          'rbt-aux-lg': bsSize === 'large' || bsSize === 'lg'
        })
      }, content) : null;
    });

    return _this;
  }

  var _proto = TypeaheadComponent.prototype;

  _proto.render = function render() {
    var _this2 = this;

    var _this$props3 = this.props,
        children = _this$props3.children,
        className = _this$props3.className,
        style = _this$props3.style;
    return React.createElement(Typeahead, _extends({}, this.props, {
      ref: function ref(instance) {
        return _this2._instance = instance;
      }
    }), function (props) {
      var auxContent = _this2._renderAux(props);

      return React.createElement("div", {
        className: cx('rbt', {
          'has-aux': !!auxContent
        }, className),
        style: _extends({
          outline: 'none',
          position: 'relative'
        }, style),
        tabIndex: -1
      }, React.createElement(Typeahead.Input, null, _this2._renderInput), React.createElement(Typeahead.Menu, null, _this2._renderMenu), auxContent, isFunction(children) ? children(props) : children);
    });
  };

  return TypeaheadComponent;
}(React.Component);

_defineProperty(TypeaheadComponent, "propTypes", propTypes);

_defineProperty(TypeaheadComponent, "defaultProps", defaultProps);

export default TypeaheadComponent;
