// @patternfly/react-table/dist/js/components/Table/examples/ActionsTable.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactTable = require('@patternfly/react-table');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var ActionsTable = function (_React$Component) {
  _inherits(ActionsTable, _React$Component);

  function ActionsTable(props) {
    _classCallCheck(this, ActionsTable);

    var _this = _possibleConstructorReturn(this, (ActionsTable.__proto__ || Object.getPrototypeOf(ActionsTable)).call(this, props));

    _this.state = {
      columns: [{ title: 'Repositories' }, 'Branches', { title: 'Pull requests' }, 'Workspaces', 'Last Commit'],
      rows: [{
        cells: ['one', 'two', 'a', 'four', 'five']
      }, {
        cells: ['a', 'two', 'k', 'four', 'five']
      }, {
        cells: ['p', 'two', 'b', 'four', 'five']
      }],
      actions: [{
        title: 'Some action',
        onClick: function onClick(event, rowId) {
          return console.log('clicked on Some action, on row: ', rowId);
        }
      }, {
        title: _react2.default.createElement(
          'div',
          null,
          'Another action'
        ),
        onClick: function onClick(event, rowId) {
          return console.log('clicked on Another action, on row: ', rowId);
        }
      }, {
        isSeparator: true
      }, {
        title: 'Third action',
        onClick: function onClick(event, rowId) {
          return console.log('clicked on Third action, on row: ', rowId);
        }
      }]
    };
    return _this;
  }

  _createClass(ActionsTable, [{
    key: 'render',
    value: function render() {
      var _state = this.state,
          columns = _state.columns,
          rows = _state.rows,
          actions = _state.actions;

      return _react2.default.createElement(
        _reactTable.Table,
        { caption: 'Actions Table', actions: actions, cells: columns, rows: rows },
        _react2.default.createElement(_reactTable.TableHeader, null),
        _react2.default.createElement(_reactTable.TableBody, null)
      );
    }
  }]);

  return ActionsTable;
}(_react2.default.Component);

exports.default = ActionsTable;
