// @patternfly/react-table/dist/js/components/Table/examples/SimpleTable.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactTable = require('@patternfly/react-table');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var SimpleTable = function (_React$Component) {
  _inherits(SimpleTable, _React$Component);

  function SimpleTable(props) {
    _classCallCheck(this, SimpleTable);

    var _this = _possibleConstructorReturn(this, (SimpleTable.__proto__ || Object.getPrototypeOf(SimpleTable)).call(this, props));

    _this.state = {
      columns: [{ title: 'Repositories' }, 'Branches', { title: 'Pull requests' }, 'Workspaces', 'Last Commit'],
      rows: [['one', 'two', 'three', 'four', 'five'], [{
        title: _react2.default.createElement(
          'div',
          null,
          'one - 2'
        ),
        props: { title: 'hover title', colSpan: 3 }
      }, 'four - 2', 'five - 2']]
    };
    return _this;
  }

  _createClass(SimpleTable, [{
    key: 'render',
    value: function render() {
      var _state = this.state,
          columns = _state.columns,
          rows = _state.rows;


      return _react2.default.createElement(
        _reactTable.Table,
        { caption: 'Simple Table', cells: columns, rows: rows },
        _react2.default.createElement(_reactTable.TableHeader, null),
        _react2.default.createElement(_reactTable.TableBody, null)
      );
    }
  }]);

  return SimpleTable;
}(_react2.default.Component);

SimpleTable.title = 'Simple Table';
exports.default = SimpleTable;
