// @patternfly/react-table/dist/js/components/Table/examples/SortableTable.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _reactTable = require('@patternfly/react-table');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var SortableTable = function (_React$Component) {
  _inherits(SortableTable, _React$Component);

  function SortableTable(props) {
    _classCallCheck(this, SortableTable);

    var _this = _possibleConstructorReturn(this, (SortableTable.__proto__ || Object.getPrototypeOf(SortableTable)).call(this, props));

    _this.state = {
      columns: [{ title: 'Repositories', transforms: [_reactTable.sortable] }, 'Branches', { title: 'Pull requests', transforms: [_reactTable.sortable] }, 'Workspaces', 'Last Commit'],
      rows: [['one', 'two', 'a', 'four', 'five'], ['a', 'two', 'k', 'four', 'five'], ['p', 'two', 'b', 'four', 'five']],
      sortBy: {}
    };
    _this.onSort = _this.onSort.bind(_this);
    return _this;
  }

  _createClass(SortableTable, [{
    key: 'onSort',
    value: function onSort(_event, index, direction) {
      var sortedRows = this.state.rows.sort(function (a, b) {
        return a[index] < b[index] ? -1 : a[index] > b[index] ? 1 : 0;
      });
      this.setState({
        sortBy: {
          index: index,
          direction: direction
        },
        rows: direction === _reactTable.SortByDirection.asc ? sortedRows : sortedRows.reverse()
      });
    }
  }, {
    key: 'render',
    value: function render() {
      var _state = this.state,
          columns = _state.columns,
          rows = _state.rows,
          sortBy = _state.sortBy;


      return _react2.default.createElement(
        _reactTable.Table,
        { caption: 'Sortable Table', sortBy: sortBy, onSort: this.onSort, cells: columns, rows: rows },
        _react2.default.createElement(_reactTable.TableHeader, null),
        _react2.default.createElement(_reactTable.TableBody, null)
      );
    }
  }]);

  return SortableTable;
}(_react2.default.Component);

exports.default = SortableTable;
