// @patternfly/react-table/dist/esm/components/Table/utils/decorators/sortable.js
import React from 'react';
import SortColumn, { SortByDirection } from '../../SortColumn';
import { css } from '@patternfly/react-styles';
import styles from '../../../../@patternfly/patternfly/components/Table/table.css.js';
import buttonStyles from '../../../../@patternfly/patternfly/components/Button/button.css.js';

export default (function (label, _ref) {
  var _ref$column$extraPara = _ref.column.extraParams,
      sortBy = _ref$column$extraPara.sortBy,
      onSort = _ref$column$extraPara.onSort,
      columnIndex = _ref.columnIndex;

  var isSortedBy = sortBy && columnIndex === sortBy.index;
  function sortClicked(event) {
    var reversedDirection = void 0;
    if (!isSortedBy) {
      reversedDirection = SortByDirection.asc;
    } else {
      reversedDirection = sortBy.direction === SortByDirection.asc ? SortByDirection.desc : SortByDirection.asc;
    }
    onSort && onSort(event, columnIndex, reversedDirection);
  }

  return {
    className: css(styles.tableSort, isSortedBy && styles.modifiers.selected),
    'aria-sort': isSortedBy ? sortBy.direction + 'ending' : 'none',
    children: React.createElement(
      SortColumn,
      {
        isSortedBy: isSortedBy,
        sortDirection: isSortedBy ? sortBy.direction : '',
        onSort: sortClicked,
        className: css(buttonStyles.button, buttonStyles.modifiers.plain)
      },
      label
    )
  };
});
