// react-bootstrap-typeahead/lib/components/Typeahead.react.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inheritsLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/inheritsLoose"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _classnames = _interopRequireDefault(require("classnames"));

var _isFunction = _interopRequireDefault(require("lodash/isFunction"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _react = _interopRequireDefault(require("react"));

var _Typeahead = _interopRequireDefault(require("../core/Typeahead"));

var _ClearButton = _interopRequireDefault(require("./ClearButton.react"));

var _Loader = _interopRequireDefault(require("./Loader.react"));

var _TypeaheadInputMulti = _interopRequireDefault(require("./TypeaheadInputMulti.react"));

var _TypeaheadInputSingle = _interopRequireDefault(require("./TypeaheadInputSingle.react"));

var _TypeaheadMenu = _interopRequireDefault(require("./TypeaheadMenu.react"));

var _utils = require("../utils");

var propTypes = {
  /**
   * Specifies the size of the input.
   */
  bsSize: _propTypes.default.oneOf(['large', 'lg', 'small', 'sm']),

  /**
   * Displays a button to clear the input when there are selections.
   */
  clearButton: _propTypes.default.bool,

  /**
   * Bootstrap 4 only. Adds the `is-invalid` classname to the `form-control`.
   */
  isInvalid: _propTypes.default.bool,

  /**
   * Indicate whether an asynchronous data fetch is happening.
   */
  isLoading: _propTypes.default.bool,

  /**
   * Bootstrap 4 only. Adds the `is-valid` classname to the `form-control`.
   */
  isValid: _propTypes.default.bool,

  /**
   * Callback for custom input rendering.
   */
  renderInput: _propTypes.default.func,

  /**
   * Callback for custom menu rendering.
   */
  renderMenu: _propTypes.default.func
};
var defaultProps = {
  clearButton: false,
  isInvalid: false,
  isLoading: false,
  isValid: false,
  renderMenu: function renderMenu(results, menuProps) {
    return (// TODO: Merged props not registering correctly.
      // $FlowFixMe
      _react.default.createElement(_TypeaheadMenu.default, (0, _extends2.default)({}, menuProps, {
        options: results
      }))
    );
  }
};

var TypeaheadComponent =
/*#__PURE__*/
function (_React$Component) {
  (0, _inheritsLoose2.default)(TypeaheadComponent, _React$Component);

  function TypeaheadComponent() {
    var _this;

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _React$Component.call.apply(_React$Component, [this].concat(args)) || this;
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "_instance", null);
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "getInstance", function () {
      return _this._instance;
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "_renderInput", function (inputProps) {
      var _this$props = _this.props,
          bsSize = _this$props.bsSize,
          isInvalid = _this$props.isInvalid,
          isValid = _this$props.isValid,
          multiple = _this$props.multiple,
          renderInput = _this$props.renderInput,
          renderToken = _this$props.renderToken;

      if ((0, _isFunction.default)(renderInput)) {
        return renderInput(inputProps);
      }

      var props = (0, _extends2.default)({}, inputProps, {
        bsSize: bsSize,
        isInvalid: isInvalid,
        isValid: isValid
      });
      return multiple ? _react.default.createElement(_TypeaheadInputMulti.default, (0, _extends2.default)({}, props, {
        renderToken: renderToken
      })) : _react.default.createElement(_TypeaheadInputSingle.default, props);
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "_renderMenu", function (results, menuProps) {
      var _this$props2 = _this.props,
          emptyLabel = _this$props2.emptyLabel,
          maxHeight = _this$props2.maxHeight,
          newSelectionPrefix = _this$props2.newSelectionPrefix,
          renderMenu = _this$props2.renderMenu,
          renderMenuItemChildren = _this$props2.renderMenuItemChildren;
      return renderMenu(results, (0, _extends2.default)({}, menuProps, {
        emptyLabel: emptyLabel,
        maxHeight: maxHeight,
        newSelectionPrefix: newSelectionPrefix,
        renderMenuItemChildren: renderMenuItemChildren
      }));
    });
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "_renderAux", function (props) {
      var bsSize = props.bsSize,
          clearButton = props.clearButton,
          disabled = props.disabled,
          isLoading = props.isLoading,
          onClear = props.onClear,
          selected = props.selected;
      var content;

      if (isLoading) {
        content = _react.default.createElement(_Loader.default, {
          bsSize: bsSize
        });
      } else if (clearButton && !disabled && selected.length) {
        content = _react.default.createElement(_ClearButton.default, {
          bsSize: bsSize,
          onClick: onClear,
          onFocus: function onFocus(e) {
            // Prevent the main input from auto-focusing again.
            e.stopPropagation();
          },
          onMouseDown: _utils.preventInputBlur
        });
      }

      return content ? _react.default.createElement("div", {
        className: (0, _classnames.default)('rbt-aux', {
          'rbt-aux-lg': bsSize === 'large' || bsSize === 'lg'
        })
      }, content) : null;
    });
    return _this;
  }

  var _proto = TypeaheadComponent.prototype;

  _proto.render = function render() {
    var _this2 = this;

    var _this$props3 = this.props,
        children = _this$props3.children,
        className = _this$props3.className,
        style = _this$props3.style;
    return _react.default.createElement(_Typeahead.default, (0, _extends2.default)({}, this.props, {
      ref: function ref(instance) {
        return _this2._instance = instance;
      }
    }), function (props) {
      var auxContent = _this2._renderAux(props);

      return _react.default.createElement("div", {
        className: (0, _classnames.default)('rbt', {
          'has-aux': !!auxContent
        }, className),
        style: (0, _extends2.default)({
          outline: 'none',
          position: 'relative'
        }, style),
        tabIndex: -1
      }, _react.default.createElement(_Typeahead.default.Input, null, _this2._renderInput), _react.default.createElement(_Typeahead.default.Menu, null, _this2._renderMenu), auxContent, (0, _isFunction.default)(children) ? children(props) : children);
    });
  };

  return TypeaheadComponent;
}(_react.default.Component);

(0, _defineProperty2.default)(TypeaheadComponent, "propTypes", propTypes);
(0, _defineProperty2.default)(TypeaheadComponent, "defaultProps", defaultProps);
var _default = TypeaheadComponent;
exports.default = _default;
