// @patternfly/react-core/dist/umd/components/Title/Title.js
"use strict";

(function (global, factory) {
  if (typeof define === "function" && define.amd) {
    define(["exports", "react", "@patternfly/react-styles", "prop-types", "../../@patternfly/patternfly/components/Title/title.css.js", "../../styles/sizes"], factory);
  } else if (typeof exports !== "undefined") {
    factory(exports, require("react"), require("@patternfly/react-styles"), require("prop-types"), require("../../@patternfly/patternfly/components/Title/title.css.js"), require("../../styles/sizes"));
  } else {
    var mod = {
      exports: {}
    };
    factory(mod.exports, global.react, global.reactStyles, global.propTypes, global.titleCss, global.sizes);
    global.undefined = mod.exports;
  }
})(void 0, function (exports, _react, _reactStyles, _propTypes, _titleCss, _sizes) {
  "use strict";

  Object.defineProperty(exports, "__esModule", {
    value: true
  });
  exports.TitleSize = undefined;

  var _react2 = _interopRequireDefault(_react);

  var _propTypes2 = _interopRequireDefault(_propTypes);

  var _titleCss2 = _interopRequireDefault(_titleCss);

  function _interopRequireDefault(obj) {
    return obj && obj.__esModule ? obj : {
      default: obj
    };
  }

  var _TitleSize;

  function _extends() {
    _extends = Object.assign || function (target) {
      for (var i = 1; i < arguments.length; i++) {
        var source = arguments[i];

        for (var key in source) {
          if (Object.prototype.hasOwnProperty.call(source, key)) {
            target[key] = source[key];
          }
        }
      }

      return target;
    };

    return _extends.apply(this, arguments);
  }

  function _objectWithoutProperties(source, excluded) {
    if (source == null) return {};

    var target = _objectWithoutPropertiesLoose(source, excluded);

    var key, i;

    if (Object.getOwnPropertySymbols) {
      var sourceSymbolKeys = Object.getOwnPropertySymbols(source);

      for (i = 0; i < sourceSymbolKeys.length; i++) {
        key = sourceSymbolKeys[i];
        if (excluded.indexOf(key) >= 0) continue;
        if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue;
        target[key] = source[key];
      }
    }

    return target;
  }

  function _objectWithoutPropertiesLoose(source, excluded) {
    if (source == null) return {};
    var target = {};
    var sourceKeys = Object.keys(source);
    var key, i;

    for (i = 0; i < sourceKeys.length; i++) {
      key = sourceKeys[i];
      if (excluded.indexOf(key) >= 0) continue;
      target[key] = source[key];
    }

    return target;
  }

  function _defineProperty(obj, key, value) {
    if (key in obj) {
      Object.defineProperty(obj, key, {
        value: value,
        enumerable: true,
        configurable: true,
        writable: true
      });
    } else {
      obj[key] = value;
    }

    return obj;
  }

  var TitleSize = exports.TitleSize = (_TitleSize = {}, _defineProperty(_TitleSize, _sizes.BaseSizes.md, _sizes.BaseSizes.md), _defineProperty(_TitleSize, _sizes.BaseSizes.lg, _sizes.BaseSizes.lg), _defineProperty(_TitleSize, _sizes.BaseSizes.xl, _sizes.BaseSizes.xl), _defineProperty(_TitleSize, _sizes.BaseSizes['2xl'], _sizes.BaseSizes['2xl']), _defineProperty(_TitleSize, _sizes.BaseSizes['3xl'], _sizes.BaseSizes['3xl']), _defineProperty(_TitleSize, _sizes.BaseSizes['4xl'], _sizes.BaseSizes['4xl']), _TitleSize);
  var propTypes = {
    /** the size of the Title  */
    size: _propTypes2.default.oneOf(Object.values(TitleSize)).isRequired,

    /** content rendered inside the Title */
    children: _propTypes2.default.node,

    /** additional classes added to the Title */
    className: _propTypes2.default.string,

    /** the heading level to use */
    headingLevel: _propTypes2.default.oneOf(['h1', 'h2', 'h3', 'h4', 'h5', 'h6']),

    /** Additional props are spread to the HeadingLevel component */
    '': _propTypes2.default.any
  };
  var defaultProps = {
    children: '',
    className: '',
    headingLevel: 'h1'
  };

  var Title = function Title(_ref) {
    var size = _ref.size,
        className = _ref.className,
        children = _ref.children,
        HeadingLevel = _ref.headingLevel,
        props = _objectWithoutProperties(_ref, ["size", "className", "children", "headingLevel"]);

    return _react2.default.createElement(HeadingLevel, _extends({}, props, {
      className: (0, _reactStyles.css)(_titleCss2.default.title, (0, _reactStyles.getModifier)(_titleCss2.default, size), className)
    }), children);
  };

  Title.propTypes = propTypes;
  Title.defaultProps = defaultProps;
  exports.default = Title;
});
