// react-bootstrap-typeahead/lib/core/Context.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.withContext = exports.MenuContext = exports.MenuContextKeys = exports.InputContext = exports.InputContextKeys = exports.TypeaheadContext = exports.TypeaheadContextKeys = void 0;

var _inheritsLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/inheritsLoose"));

var _extends3 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _pick = _interopRequireDefault(require("lodash/pick"));

var _createReactContext = _interopRequireDefault(require("create-react-context"));

var _react = _interopRequireDefault(require("react"));

/* istanbul ignore next */
var arrToObj = function arrToObj(keys) {
  return keys.reduce(function (obj, k) {
    var _extends2;

    return (0, _extends3.default)({}, obj, (_extends2 = {}, _extends2[k] = undefined, _extends2));
  }, {});
};
/**
 * $Pick utility
 * https://github.com/facebook/flow/issues/3367
 */

/* eslint-disable flowtype/generic-spacing */


/* eslint-enable flowtype/generic-spacing */
var TypeaheadContextKeys = ['activeIndex', 'id', 'initialItem', 'onActiveItemChange', 'onAdd', 'onInitialItemChange', 'onMenuItemClick', 'selectHintOnEnter'];
exports.TypeaheadContextKeys = TypeaheadContextKeys;
var TypeaheadContext = (0, _createReactContext.default)({});
exports.TypeaheadContext = TypeaheadContext;
var InputContextKeys = ['activeIndex', 'disabled', 'id', 'inputProps', 'inputRef', 'isFocused', 'isMenuShown', 'labelKey', 'multiple', 'onBlur', 'onChange', 'onFocus', 'onKeyDown', 'onRemove', 'placeholder', 'selected'];
exports.InputContextKeys = InputContextKeys;
var InputContext = (0, _createReactContext.default)({});
exports.InputContext = InputContext;
var MenuContextKeys = ['align', 'dropup', 'flip', 'id', 'labelKey', 'onMenuToggle', 'positionFixed', 'referenceElement', 'results', 'text'];
exports.MenuContextKeys = MenuContextKeys;
var MenuContext = (0, _createReactContext.default)({});
exports.MenuContext = MenuContext;

var withContext = function withContext(Component, values) {
  // Note: Use a class instead of function component to support refs.

  /* eslint-disable-next-line react/prefer-stateless-function */
  return (
    /*#__PURE__*/
    function (_React$Component) {
      (0, _inheritsLoose2.default)(_class, _React$Component);

      function _class() {
        return _React$Component.apply(this, arguments) || this;
      }

      var _proto = _class.prototype;

      _proto.render = function render() {
        var _this = this;

        return _react.default.createElement(TypeaheadContext.Consumer, null, function (context) {
          return _react.default.createElement(Component, (0, _extends3.default)({}, _this.props, (0, _pick.default)(context, values)));
        });
      };

      return _class;
    }(_react.default.Component)
  );
};

exports.withContext = withContext;
