// react-bootstrap-typeahead/lib/core/TypeaheadInput.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutPropertiesLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutPropertiesLoose"));

var _classnames = _interopRequireDefault(require("classnames"));

var _react = _interopRequireDefault(require("react"));

var _Context = require("./Context");

var _utils = require("../utils");

var TypeaheadInput = function TypeaheadInput(_ref) {
  var children = _ref.children;
  return _react.default.createElement(_Context.InputContext.Consumer, null, function (context) {
    var _cx;

    var activeIndex = context.activeIndex,
        id = context.id,
        inputProps = context.inputProps,
        isFocused = context.isFocused,
        isMenuShown = context.isMenuShown,
        labelKey = context.labelKey,
        multiple = context.multiple,
        onFocus = context.onFocus,
        onRemove = context.onRemove,
        placeholder = context.placeholder,
        selected = context.selected,
        otherProps = (0, _objectWithoutPropertiesLoose2.default)(context, ["activeIndex", "id", "inputProps", "isFocused", "isMenuShown", "labelKey", "multiple", "onFocus", "onRemove", "placeholder", "selected"]); // Add a11y-related props.

    var props = (0, _extends2.default)({}, inputProps, otherProps, {
      'aria-activedescendant': activeIndex >= 0 ? (0, _utils.getMenuItemId)(id, activeIndex) : '',
      'aria-autocomplete': multiple ? 'list' : 'both',
      'aria-expanded': isMenuShown,
      'aria-haspopup': 'listbox',
      'aria-owns': isMenuShown ? id : '',
      autoComplete: inputProps.autoComplete || 'off',
      // Re-open the menu, eg: if it's closed via ESC.
      onClick: onFocus,
      onFocus: onFocus,
      placeholder: selected.length ? null : placeholder,
      // Comboboxes are single-select by definition:
      // https://www.w3.org/TR/wai-aria-practices-1.1/#combobox
      role: 'combobox',
      type: inputProps.type || 'text'
    });
    var className = props.className || '';

    if (multiple) {
      props = (0, _extends2.default)({}, props, {
        'aria-expanded': undefined,
        inputClassName: className,
        labelKey: labelKey,
        onRemove: onRemove,
        role: undefined,
        selected: selected
      });
    }

    props.className = (0, _classnames.default)((_cx = {}, _cx[className] = !multiple, _cx.focus = isFocused, _cx));
    return children(props);
  });
};

var _default = TypeaheadInput;
exports.default = _default;
