// @patternfly/react-table/dist/js/components/Table/Table.docs.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _reactTable = require('@patternfly/react-table');

var _SimpleTable = require('./examples/SimpleTable');

var _SimpleTable2 = _interopRequireDefault(_SimpleTable);

var _SortableTable = require('./examples/SortableTable');

var _SortableTable2 = _interopRequireDefault(_SortableTable);

var _SelectableTable = require('./examples/SelectableTable');

var _SelectableTable2 = _interopRequireDefault(_SelectableTable);

var _ActionsTable = require('./examples/ActionsTable');

var _ActionsTable2 = _interopRequireDefault(_ActionsTable);

var _CellHeader = require('./examples/CellHeader');

var _CellHeader2 = _interopRequireDefault(_CellHeader);

var _CompactTable = require('./examples/CompactTable');

var _CompactTable2 = _interopRequireDefault(_CompactTable);

var _WidthTable = require('./examples/WidthTable');

var _WidthTable2 = _interopRequireDefault(_WidthTable);

var _CollapsibleTable = require('./examples/CollapsibleTable');

var _CollapsibleTable2 = _interopRequireDefault(_CollapsibleTable);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = {
  title: 'Table',
  components: {
    Table: _reactTable.Table,
    TableBody: _reactTable.TableBody,
    TableHeader: _reactTable.TableHeader
  },
  variablesRoot: 'pf-c-table',
  examples: [{ component: _SimpleTable2.default, title: 'Simple Table' }, { component: _SortableTable2.default, title: 'Sortable Table' }, { component: _SelectableTable2.default, title: 'Selectable Table' }, { component: _ActionsTable2.default, title: 'Actions Table' }, { component: _CellHeader2.default, title: 'First cell as Header' }, { component: _CompactTable2.default, title: 'Compact Table' }, { component: _WidthTable2.default, title: 'Table with Width Modifiers' }, { component: _CollapsibleTable2.default, title: 'Collapsible table' }]
};
